### PR TITLE
Logtest

### PR DESCRIPTION
**Pull request by alicerobson**
This pull request implements
(1) addition into "make test" of runs to check the physics outputs for CLIC and CMS detectors. This is carried out by two scripts pdebug_CMS.sh etc
(2) CMS/Clic can be chosen as a command line option and PythiaConnecter has a new process event to enable this to work OK.
(3) Initial implementation of  a "papaslog" version of ATS logging which allows logging to be attached into Gaudi logging. A little clunky and likely to change but it is a start
(4) clang formatting (I know... should have done this earlier, sorry)

### CODE REVIEW

File: papas/detectors/clic/src/ClicTracker.cpp [hunk 1]
```diff
@@ -48,30 +48,29 @@ double ClicTracker::resolution(const Track& track) const {
...
+  // will do for now
+  double pt = ptc.p3().Pt();
+  // matching the resmap defined above.
+  double theta = fabs(ptc.theta()) * 180 / M_PI;
+  for (const auto& v : m_resMap) {
+    if (theta < v.first) return sigmaPtOverPt2(v.second.first, v.second.second, pt) * pt;
   }
-  
-}// end namespace papas
+  throw "tracker particle resolution not found";
```
Comment (cbernet) [inline papas/detectors/clic/src/ClicTracker.cpp]: Naive question: shouldn't that be a PAPASLOG_ERROR?
Maybe review the use of throw to see which ones should be a PAPASLOG_ERROR

File: examples/PythiaConnector.cpp [hunk 2]
```diff
@@ -43,7 +43,7 @@ PythiaConnector::PythiaConnector(const char* fname) : m_store(podio::EventStore(
   if (access(fname, F_OK) != -1) {
     m_reader.openFile(fname);
   } else {
-    throw "File not found";
+    PAPASLOG_ERROR("File not found");
```
Comment (cbernet) [inline examples/PythiaConnector.cpp]: Is PAPASLOG_ERROR doing the throw?
If not are you sure we should remove the throw?

File: papas/detectors/clic/src/ClicTracker.cpp [hunk 3]
```diff
@@ -48,30 +48,29 @@ double ClicTracker::resolution(const Track& track) const {
...
+  return 0;
+}
+
+bool ClicTracker::electronAcceptance(double mag, const Track& track) const {
```
Comment (cbernet) [inline papas/detectors/clic/src/ClicTracker.cpp]: Why is that part of ClicTracker? 
in python, it's part of the CLIC detector. 
The electron acceptance is not determined only by the tracker in principle

File: papas/detectors/clic/src/ClicTracker.cpp [hunk 4]
```diff
@@ -48,30 +48,29 @@ double ClicTracker::resolution(const Track& track) const {
...
-    }
-    throw "tracker particle resolution not found";
-    return 0;
-  }
-  
-  bool ClicTracker::electronAcceptance(double mag, const Track& track) const {
-    if ( track.p3().Pt() > mag && fabs(track.theta()) < m_thetaMax)
-      return rootrandom::Random::uniform(0, 1) > 0.95;
-    return false;
+double ClicTracker::particleResolution(const Particle& ptc) const {
```
Comment (cbernet) [inline papas/detectors/clic/src/ClicTracker.cpp]: Why particleResolution and not resolution as in python?

Comment (alicerobson) [reply papas/detectors/clic/src/ClicTracker.cpp]: Ah yes! I had to create a second function as the resolution function can receive either a particle or a track in python. But you are right the extra function can still have the same name (thanks)

File: papas/detectors/clic/src/ClicTracker.cpp (hunk 1 above)
Comment (alicerobson) [reply papas/detectors/clic/src/ClicTracker.cpp]: I think this is OK. The idea is that papas can throw during an event and then  this should be caught and the event discarded ie there needs to be somewhere that centrally manages any issues in processing a specific event. So I judged that it is OK to throw to the centralized area and then issue a formal error such as PAPASLOG_ERROR  at this point.  This means we have most of the calls to PAPASLOG_ERROR just after the throw is caught. In fact in gaudi once the throw is caught the error can go direct to its logger. I hope that makes sense :-). It seems to work.

File: papas/detectors/clic/src/ClicTracker.cpp (hunk 3 above)
Comment (alicerobson) [reply papas/detectors/clic/src/ClicTracker.cpp]: OK, have put it back in the tracker.

File: examples/PythiaConnector.cpp (hunk 2 above)
Comment (alicerobson) [reply examples/PythiaConnector.cpp]: The throw was already removed (its in pink with a - sign &  not green with a +) :-)

**Pull request closed without merging**